### PR TITLE
qemu.tests.virtual_nic_private:adjust launch 3 VMS's memory.

### PR DESCRIPTION
**Pull request by weliao**
ID: 1401810

Signed-off-by: Wei Liao <weliao@redhat.com>

### CODE REVIEW

File: qemu/tests/virtual_nic_private.py [hunk 1]
```diff
@@ -47,17 +50,22 @@ def data_mon(session, cmd, timeout):
     mon_process_timeout = int(params.get("mon_process_timeout", "1200"))
     sessions = []
     addresses = []
-    vms = []
+    vms = params["vms"].split()
```
Comment (xutian) [inline qemu/tests/virtual_nic_private.py]: param['mem'] =  (host_free_memory * 1.2  /  len(vms)) / 64  + 1)  *  64

How about limited guest memory size to 64 times

File: qemu/tests/virtual_nic_private.py [hunk 2]
```diff
@@ -47,17 +50,22 @@ def data_mon(session, cmd, timeout):
...
-    for vm in vms:
+    for vm_name in vms:
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
         vm.verify_alive()
         sessions.append(vm.wait_for_login(timeout=timeout))
         addresses.append(vm.get_address())
-    mon_session = vms[2].wait_for_login(timeout=timeout)
-    mon_macaddr = vms[2].get_mac_address()
+
```
Comment (xutian) [inline qemu/tests/virtual_nic_private.py]: if session not closed, it looks no diff between your code

File: qemu/tests/virtual_nic_private.py (hunk 1 above)
Comment (weliao) [inline qemu/tests/virtual_nic_private.py]: @xutian  Updated, could you help check it again? thanks

File: qemu/tests/virtual_nic_private.py (hunk 2 above)
Comment (weliao) [inline qemu/tests/virtual_nic_private.py]: need create a new session here and vms change for me.

File: qemu/tests/virtual_nic_private.py [hunk 3]
```diff
@@ -47,17 +50,23 @@ def data_mon(session, cmd, timeout):
     mon_process_timeout = int(params.get("mon_process_timeout", "1200"))
     sessions = []
     addresses = []
-    vms = []
+    vms = params["vms"].split()
+    host_cmd_output = utils.system_output("sync && echo 3 > /proc/sys/vm/drop_caches")
```
Comment (CongLi) [inline qemu/tests/virtual_nic_private.py]: need ignore status for this cmd.

**Comment by CongLi** [comment]
@weliao 

Could you please give more details in the commit log about the fix?
I don't know why about your fix.

**Comment by CongLi** [comment]
@weliao 

Could you check if https://github.com/avocado-framework/avocado-vt/pull/830 has been fixed your problem?

Thanks.

**Comment by weliao** [comment]
@CongLi yes, his PR is better, I will close this PR.

**Pull request closed without merging**